🎉 ♻️ Initial commit of refactor

znatty22 opened the pull request:
Some changes:
- Use pull request endpoint to construct feature list instead of commits endpoint
- Categorize features by emoji (don't assume PRs have labels)
- Allow creation of patch versions
- Basic code structure refactor including:
    - Refactor click CLI so that we can add more commands in future (may want to add cmds to create release PR, create a tag, etc)

znatty22 committed:
:tada: Initial commit

fiendish committed:
Update README.md

dankolbman commented on kf_release_maker/release_maker.py:
```diff
@@ -0,0 +1,276 @@
+        notes += '\n\n## Features\n\n'
+        notes += '### Summary\n\n'
+        notes += f'Feature Emojis: {emoji_counts} \n'
+        notes += f'Feature Categories: {category_counts} \n\n'
```
What was wrong with using the labels here?

znatty22 replied:
Not all of our PRs have labels and it seems like we already use emojis to do some categorization of changes. I decided to bucket the emojis into categories and use those instead.  Also, I think we should talk about how we use emojis and labels and how to avoid redundancy

dankolbman replied:
Which prs do not have labels? All PRs that I'm familiar with do and it's also mentioned in the [handbook](https://kids-first.github.io/kf-developer-handbook/guides/pull_requests.html).
Emojis do often correlate with the label, but not all the time.

dankolbman commented on kf_release_maker/release_maker.py:
```diff
@@ -0,0 +1,276 @@
+from urllib.parse import urlencode
+from pprint import pformat
+
+import pandas
```
Do we really need pandas here?
It's quite a heavy dependency for a small script.

znatty22 replied:
Maybe not, just thought it was easy to use for what I wanted

dankolbman reviewed (commented):
Thanks for cleaning this up for me.
What were the issues you ran into with it?

It's a little hard for me to tell what was changed, did you track the move with `git mv`?

fiendish committed:
LABEL IS WRONG, LOL

gitmoji says that releases are supposed to use "bookmark". Label means something else.

fiendish commented on kf_release_maker/release_maker.py:
```diff
@@ -0,0 +1,276 @@
+        notes = f"# {title} Release {self.release_version}"
+        notes += '\n\n## Features\n\n'
+        notes += '### Summary\n\n'
+        notes += f'Feature Emojis: {emoji_counts} \n'
```
```suggestion
        notes += f'Feature Emojis: {emoji_counts} \n\n'
```

fiendish commented:
Why is this sitting in a draft?

znatty22 commented:
> Why is this sitting in a draft?

@fiendish Idk 🤷 I was going to clean up the code a bit and replace pandas with plain Python, but didn't get around to it.

fiendish commented:
> Use pull request endpoint to construct feature list instead of commits endpoint

Doing this means you lose all your items if you migrate the repository because PRs are github metadata not stored in the repo.

znatty22 commented:
But @fiendish I don't think we're moving away from Github anytime soon? And the other major version control web app is Gitlab which also supports the concept of Pull Requests.
